src/open_r1/rewards.py

```diff
@@ -0,0 +1,79 @@
+"""Reward functions for GRPO training."""
+
+import re
+from math_verify import LatexExtractionConfig, parse, verify
+from latex2sympy2_extended import NormalizationConfig
+
+
+def accuracy_reward(completions, solution, **kwargs):
+    """Reward function that checks if the completion is the same as the ground truth."""
+    contents = [completion[0]["content"] for completion in completions]
+    rewards = []
+    for content, sol in zip(contents, solution):
+        gold_parsed = parse(
+            sol,
+            extraction_mode="first_match",
+            extraction_config=[LatexExtractionConfig()],
+        )
+        if len(gold_parsed) != 0:
+            # We require the answer to be provided in correct latex (no malformed operators)
+            answer_parsed = parse(
+                content,
+                extraction_config=[
+                    LatexExtractionConfig(
+                        normalization_config=NormalizationConfig(
+                            nits=False,
+                            malformed_operators=False,
+                            basic_latex=True,
+                            equations=True,
+                            boxed=True,
+                            units=True,
+                        ),
+                        # Ensures that boxed is tried first
+                        boxed_match_priority=0,
+                        try_extract_without_anchor=False,
+                    )
+                ],
+                extraction_mode="first_match",
+            )
+            # Reward 1 if the content is the same as the ground truth, 0 otherwise
+            reward = float(verify(answer_parsed, gold_parsed))
+        else:
+            # If the gold solution is not parseable, we reward 1 to skip this example
+            reward = 1.0
```

Comment:
I don't follow: why do we give the same reward in the "non parseable" case and the correct case?

Reply:
The function was taken as is: [link](https://github.com/huggingface/open-r1/pull/144/files#diff-aa2b823b9a08fc45ead8f8707f0b0ee95f40940d85883ee1fe7edadf8a4c2b0bL83)

But, I speculate that:

1. It's rare that the ground truth dataset is not parseable.
2. The downstream GRPO trainer currently doesn't support ignoring the samples.

Reply:
shouldn't we give it 0 weight instead then?

Reply:
It's completely equivalent, the reward is normalized per group.

Reply:
Yes, I was just thinking of it. 

I guess it is guaranteed that group would contain only the completions for that particular unparseable solution, right? Because if the group would contain others, the normalization wouldnt work properly. Does it make sense?

Reply:
Nevertheless, I can come back to it in the next PR

Reply:
I'd just made an issue about this and was reading this thread to see if it was related. I thought it was odd because the odds were like 0.19, 0.20, etc., so rewarding 1 for a pair where we couldn't find the gold solution felt off.

https://github.com/huggingface/open-r1/issues/159

Reply:
> We wouldn't want to give it 0 weight either since that could be a penalty or even a net reward depending on the other rewards in the group

That's not right. Gold is shared in the group. If it's not parsable for one sample, then it's not parsable for the all group.

Reply:
Oh OK, I see now.